src/autorest-core/lib/message.ts

```diff
@@ -30,7 +30,10 @@ export interface Message {
   Key?: Iterable<string>;
   Details?: any;
   Text: string;
+
+  // injected or modified by core
   Source?: Array<SourceLocation>;
   Range?: Iterable<Range>;
   Plugin?: string;
+  FormattedMessage?: string;
```

Comment:
interesting point! so to me, the only difference between "details" (which is essentially free form) and the main thingy is whether the property applies to most messages or not. having an info url sounds awefully useful and something one could encourage for most messages (I'd say comparable to source location and such), so I'd be cool with adding it to the main type 🙂 be aware, though, that the JSON formatted messages (used in the specs CI) just serialize out the details, so if it's supposed to show up there, we either gotta modify the formatter or also add it to the details... hmmm